File: 2486-most-frequent-even-element/2486-most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        freq= {}
        for num in nums:
            if num %  2 == 0:
                if num in freq:
                    freq[num] += 1
                else:
                    freq[num] = 1
        count = 0
        res = -1
        for key in freq:
            if freq[key] > count:
                count = freq[key]
                res = key
            elif freq[key] == count:
                if res > key:
                    res = key
        return res

File: 2486-most-frequent-even-element/2486-most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        HashMap<Integer, Integer> seen = new HashMap<>();
        for(int i= 0; i< nums.length; i++){
            if(nums[i] % 2 == 0){
                seen.put(nums[i], seen.getOrDefault(nums[i], 0) + 1);
            }
        }
        int cnt = 0;
        int res = -1;
        for(int ele: seen.keySet()){
            if(seen.get(ele) > cnt){
                cnt = seen.get(ele);
                res = ele;
            }
            else if(seen.get(ele) == cnt){
                if(res > ele){
                    res = ele;
                }
            }
        }
        return res;
    }
}